components/post_markdown/post_markdown.jsx

```diff
@@ -63,17 +65,25 @@ export default class PostMarkdown extends React.PureComponent {
         const channelNamesMap = this.props.post && this.props.post.props && this.props.post.props.channel_mentions;
 
         let {message} = this.props;
-        const {post} = this.props;
-        const options = {...this.props.options};
+        const {post, allMentionKeys, mentionKeysWithoutGroups, options} = this.props;
 
         this.props.pluginHooks.forEach((o) => {
             if (o && o.hook && post) {
                 message = o.hook(post, message);
             }
         });
 
+        let mentionKeys = [];
         if (post && post.props) {
-            options.mentionHighlight = !post.props.mentionHighlightDisabled;
+            mentionKeys = allMentionKeys;
+
+            if (post.props.disable_group_highlight) {
+                mentionKeys = mentionKeysWithoutGroups;
+            }
+
+            if (post.props.mentionHighlightDisabled) {
+                mentionKeys = mentionKeys.filter((value) => !['@all', '@channel', '@here'].includes(value.key));
```

Comment:
This is going to cause `Markdown` to re-render whenever this does because `filter` creates a new array each time render is called. To solve that, we could memoize the final list of `mentionKeys` either by creating a selector in `componentspost_markdown/index.js` or we could use add a method to this component that returns this result, but it's wrapped with `memoizeResult` from `mattermost-redux/utils/helpers`.

Reply:
Done! Thanks :)